hydra-node/src/Hydra/Events/Rotation.hs

```diff
@@ -0,0 +1,104 @@
+module Hydra.Events.Rotation where
+
+import Hydra.Prelude
+
+import Conduit (MonadUnliftIO)
+import Control.Concurrent.Class.MonadSTM (newTVarIO, readTVarIO, writeTVar)
+import Hydra.Chain.ChainState (ChainStateType, IsChainState)
+import Hydra.Events (EventSink (..), EventSource (..), HasEventId, LogId, StateEvent (..), getEvents)
+import Hydra.HeadLogic (StateChanged (Checkpoint), aggregate)
+import Hydra.HeadLogic.State (HeadState (..), IdleState (..))
+
+newtype RotationConfig = RotateAfter Natural
+
+-- | An EventSource and EventSink combined
+type EventStore e m = (EventSource e m, EventSink e m)
+
+type Checkpointer e = [e] -> e
+
+-- | Creates an event store that rotates according to given config and 'Checkpointer'.
+mkRotatedEventStore ::
+  (HasEventId e, MonadSTM m, MonadUnliftIO m) =>
+  RotationConfig ->
+  Checkpointer e ->
+  LogId ->
+  EventStore e m ->
+  m (EventStore e m)
+mkRotatedEventStore config checkpointer logId eventStore = do
```

Comment:
Could: call this newRotatedEventStore

I often like to call monadic functions that create stuff `newXXX` or `createXXX` while `mkXXX` is something non-monadic.

```suggestion
newRotatedEventStore config checkpointer logId eventStore = do
```